helm/hpcc/templates/_helpers.tpl

```diff
@@ -429,6 +429,34 @@ Pass in dict with root and secretsCategories
 {{- end -}}
 {{- end -}}
 
+{{/*
+Generate Prometheus scrape annotations
+Enables selfdiscovery of metrics service on configured path/port
+Requires metrics.sinks[type=prometheus]
+Pass in metric config
+*/}}
+{{- define "hpcc.addPrometheusScrapeAnnotations" -}}
+{{- if hasKey . "sinks" }}
+ {{ range $sink := .sinks -}}
+  {{ range $sinkattindex, $sinkatts := $sink -}}
+   {{- if and  (eq $sinkattindex "type") }}
+    {{- if eq $sinkatts "prometheus" }}
+     {{- if hasKey $sink "settings" }}
+      {{- if hasKey $sink.settings "autodiscovery" }}
+       {{- if eq $sink.settings.autodiscovery true }}
```

Comment:
This looks like a case where 'dig' would be useful.
I hadn't used before, but just tested it, can express the above as:
```
{{- if (dig "settings" "autodiscovery" false $sink) }}
```
i.e. all as:
```
{{ range $sink := .sinks -}}
 {{- if eq (get $sink "type") "prometheus" }}
  {{- if (dig "settings" "autodiscovery" false $sink) }}
        prometheus.io/scrape: 'true'
        prometheus.io/path: {{ default "/metrics" $sink.settings.path }}
        prometheus.io/port: {{ default 8767 $sink.settings.port | quote }}
   {{ end }}
  {{ end }}
{{ end}}
```

Reply:
That's much cleaner, thanks for the pointers. 
The only problem is dig apparently doesn't show until more recent versions of HELM (I'm on v3.2.4 and apparently dig is not available).
Do we have a stated minimum helm version supported?
Updated helm and applied dig successfully

Reply:
hm, I'm not sure re. minimal version. I didn't know that dig was relatively new.
Unfortunately, I think it may be too new, and we should avoid for now. I say that, because I just checked what version comes by default in the Azure portal shell, and it's v3.4.0 that doesn't support dig :(

Reply:
Agreed. It's too bad, I like the streamlined optics of the function. Will revert.